### PR TITLE
[ELY-2509] Move HttpAuthenticator#isAuthenticated method to Authentic…

### PR DESCRIPTION
**Pull request by keshav-redhat**
https://issues.redhat.com/browse/ELY-2509

### CODE REVIEW

File: http/base/src/main/java/org/wildfly/security/http/HttpAuthenticator.java [hunk 1]
```diff
@@ -311,6 +308,10 @@ private class AuthenticationExchange extends BaseHttpServerRequest implements Ht
             super(httpExchangeSpi);
         }
 
+        private boolean isAuthenticated() {
+            return authenticated;
```
Comment (skyllarr) [inline http/base/src/main/java/org/wildfly/security/http/HttpAuthenticator.java:312]: @keshav-redhat Thank you for the PR! 

Just a minor, it seems that the variable `authenticated` can be moved to the `AuthenticationExchange` class as well

**Comment by skyllarr** [comment]
@keshav-redhat Just a minor, please squash the 2 commits into 1 commit since this is a related change. You can watch the last video about squashing [from our FAQ](https://fjuma.github.io/wildfly-elytron/blog/frequently-asked-questions-new-contributors/)  to see how it can be done

**Review by skyllarr — APPROVED** [review]

**Comment by skyllarr** [comment]
@keshav-redhat Thanks!

**Review by fjuma — APPROVED** [review]
Thanks @keshav-redhat!

**Pull request merged**